python/lsst/ip/diffim/subtractImages.py

```diff
@@ -0,0 +1,535 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+
+import lsst.afw.image
+import lsst.afw.math
+import lsst.geom
+from lsst.ip.diffim.utils import getPsfFwhm
+from lsst.meas.algorithms import ScaleVarianceTask
+import lsst.pex.config
+import lsst.pipe.base
+from lsst.pipe.base import connectionTypes
+
+__all__ = ["AlardLuptonSubtractConfig", "AlardLuptonSubtractTask"]
+
+_dimensions = ("instrument", "visit", "detector")
+_defaultTemplates = {"coaddName": "deep", "fakesType": ""}
+
+
+class SubtractInputConnections(lsst.pipe.base.PipelineTaskConnections,
+                               dimensions=_dimensions,
+                               defaultTemplates=_defaultTemplates):
+    template = connectionTypes.Input(
+        doc="Input warped template to subtract.",
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="ExposureF",
+        name="{fakesType}{coaddName}Diff_templateExp"
+    )
+    science = connectionTypes.Input(
+        doc="Input science exposure to subtract from.",
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="ExposureF",
+        name="{fakesType}calexp"
+    )
+    sources = connectionTypes.Input(
+        doc="Sources measured on the science exposure; "
+            "used to select sources for making the matching kernel.",
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="SourceCatalog",
+        name="{fakesType}src"
+    )
+
+
+class SubtractImageOutputConnections(lsst.pipe.base.PipelineTaskConnections,
+                                     dimensions=_dimensions,
+                                     defaultTemplates=_defaultTemplates):
+    difference = connectionTypes.Output(
+        doc="Result of subtracting convolved template from science image.",
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="ExposureF",
+        name="{fakesType}{coaddName}Diff_differenceTempExp",
+    )
+    matchedTemplate = connectionTypes.Output(
+        doc="Warped and PSF-matched template used to create `subtractedExposure`.",
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="ExposureF",
+        name="{fakesType}{coaddName}Diff_matchedExp",
+    )
+
+
+class AlardLuptonSubtractConnections(SubtractInputConnections, SubtractImageOutputConnections):
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
```

Comment:
I think you don't need this init at all. Just put `pass` under the class definition.